waku/v2/utils/peers.nim

```diff
@@ -112,7 +112,49 @@ func getTransportProtocol(typedR: TypedRecord): Option[IpTransportProtocol] =
 
 ## Parses a fully qualified peer multiaddr, in the
 ## format `(ip4|ip6)/tcp/p2p`, into dialable PeerInfo
-proc parseRemotePeerInfo*(address: string): RemotePeerInfo {.raises: [Defect, ValueError, LPError].}=
+proc parsePeerInfo*(address: string):
+                    Result[RemotePeerInfo, string] =
+  var
+    nwPart, tcpPart, p2pPart, wsPart, wssPart: MultiAddress
+
+  try:
+    let multiAddr = MultiAddress.init(address).tryGet()
+    for addrPart in multiAddr.items():
+      case addrPart[].protoName()[]
+      # All protocols listed here: https://github.com/multiformats/multiaddr/blob/b746a7d014e825221cc3aea6e57a92d78419990f/protocols.csv
+      of "ip4", "ip6", "dns", "dnsaddr", "dns4", "dns6":
+        nwPart = addrPart.tryGet()
+      of "tcp":
+        tcpPart = addrPart.tryGet()
+      of "p2p":
+        p2pPart = addrPart.tryGet()
+      of "ws":
+        wsPart = addrPart.tryGet()
+      of "wss":
+        wssPart = addrPart.tryGet()
+
+    # nim-libp2p dialing requires remote peers to be initialised with a peerId and a wire address
+    let
+      peerIdStr = p2pPart.toString()[].split("/")[^1]
+      wireAddr = nwPart & tcpPart & wsPart & wssPart
+
+    if not wireAddr.validWireAddr():
+      return err("Invalid node multiaddress")
+
+    return ok(RemotePeerInfo.init(peerIdStr, @[wireAddr]))
+  except CatchableError:
+    return err(getCurrentExceptionMsg())
+
+## Parses a fully qualified peer multiaddr, in the
+## format `(ip4|ip6)/tcp/p2p`, into dialable PeerInfo
+## Notice that we deprecated this function in favor
+## of 'parsePeerInfo' which does the same but just
+## returs a Result[...] instead of raising an exception.
+proc parseRemotePeerInfo*(address: string):
+          RemotePeerInfo
+            {.raises: [Defect, ValueError, LPError],
+            deprecated: "Raises exceptions which is difficult to maintain".}=
```

Comment:
Something I would do to avoid having duplicate functionality and to test that the function is working as expected is: 

> Call inside of the deprecated method the result method, and in the case, the result returned is an error, we raise a `ValueError`. 

With this approach, the deprecated method acts as a shim to the new result-returning implementation and we can detect with the tests if we missed anything (e.g., an unexpected exception) on the new implementation.

Reply:
Sorry I misunderstood this recommendation and I made a wider refactoring. With that I think the error cases are covered better and it gives better feedback.

Of course, if you don't agree with the changes I will roll them back and just call the "return" function inside the "exception" one as you stated in the comment :)